File: forge/src/main/java/org/valkyrienskies/clockwork/forge/content/curiosities/tools/gravitron/tool/GrabssembleTool.java
package org.valkyrienskies.clockwork.forge.content.curiosities.tools.gravitron.tool;


import org.valkyrienskies.clockwork.ClockworkPackets;
import org.valkyrienskies.clockwork.forge.content.curiosities.tools.gravitron.GravitronGrabPacket;

public class GrabssembleTool extends GravitronToolBase {

    @Override
    public boolean handleRightClick() {
        updateTargetPos();
        ClockworkPackets.sendToServer(new GravitronGrabPacket(clickedPos, clickedLocation, GRABSSEMBLE));
        return true;
    }

    @Override
    public boolean handleMouseWheel(double delta) {
        return false;
    }
}

File: common/src/main/kotlin/org/valkyrienskies/clockwork/content/curiosities/tools/gravitron/tool/GrabssembleTool.kt
package org.valkyrienskies.clockwork.content.curiosities.tools.gravitron.tool

import org.valkyrienskies.clockwork.ClockworkPackets.Companion.sendToServer
import org.valkyrienskies.clockwork.content.curiosities.tools.gravitron.GravitronGrabPacket

class GrabssembleTool : GravitronToolBase() {

    override fun handleRightClick(): Boolean {
        updateTargetPos()
        if (clickedLocation != null && clickedPos != null) {
            sendToServer(GravitronGrabPacket(clickedPos!!, clickedLocation!!, GRABSSEMBLE))
        }
        return true
    }
}